Sources/OAuth2Client/NXOAuth2Connection.m

```diff
@@ -470,8 +470,16 @@ - (void)connectionDidFinishLoading:(NSURLConnection *)connection;
                 }
             }
             if (authenticateHeader
-                && [authenticateHeader rangeOfString:@"invalid_token"].location != NSNotFound) {
```

Comment:
I would like to keep `invalid_token` to ensure we don't break existing apps that rely on that behavior. Since the lib is not (final) RFC compliant anyway this should not be a a problem. But adding `invalid_grant` seems correct.